component/http/component.go

```diff
@@ -190,6 +192,18 @@ func (cb *Builder) WithWriteTimeout(wt time.Duration) *Builder {
 	return cb
 }
 
+// WithShutdownGracePeriod sets the Shutdown Grace Period for the HTTP component.
+func (cb *Builder) WithShutdownGracePeriod(gp time.Duration) *Builder {
+	if gp <= 0*time.Second {
+		cb.errors = append(cb.errors, errors.New("negative or zero shutdown grace period provided"))
+	} else {
+		log.Infof("setting shutdown grace period")
+		cb.httpWriteTimeout = gp
```

Comment:
wrong assignment

Reply:
yes, was a typo, fixed in https://github.com/beatlabs/patron/pull/247/commits/49ebea3fe396099e86db71d7bd70944918f95600